controller/api/public/stat_summary.go

```diff
@@ -237,6 +264,93 @@ func (s *grpcServer) k8sResourceQuery(ctx context.Context, req *pb.StatSummaryRe
 	return resourceResult{res: &rsp, err: nil}
 }
 
+func (s *grpcServer) trafficSplitResourceQuery(ctx context.Context, req *pb.StatSummaryRequest) resourceResult {
+	requestedResource := req.GetSelector().GetResource()
+	objects, err := s.k8sAPI.GetObjects(requestedResource.Namespace, requestedResource.Type, requestedResource.Name)
+	if err != nil {
+		return resourceResult{res: nil, err: err}
+	}
+
+	var tsK8sInfo []*trafficSplitStats
+	tsMetrics := make(map[tsKey]*pb.BasicStats)
+
+	for _, object := range objects {
+		ts, ok := object.(*v1alpha1.TrafficSplit)
+		if !ok {
+			return resourceResult{res: nil, err: errors.New("Could not cast to trafficsplit")}
+		}
+
+		tsName := ts.ObjectMeta.Name
+		apex := ts.Spec.Service
+		backends := ts.Spec.Backends
+		tsStats := &trafficSplitStats{apex: apex, leaves: []leaf{}}
+		tsStats.name = tsName
+		tsStats.namespace = ts.ObjectMeta.Namespace
+
+		for _, returnedLeaf := range backends {
+			name := fmt.Sprint(returnedLeaf.Service)
+			weight := fmt.Sprint(returnedLeaf.Weight.String())
+
+			tsStats.leaves = append(tsStats.leaves, leaf{
+				leaf:   name,
+				weight: weight})
+		}
+
+		if !req.SkipStats {
+			tsBasicStats, err := s.getTrafficSplitMetrics(ctx, req, tsStats, req.TimeWindow)
+			if err != nil {
+				return resourceResult{res: nil, err: err}
+			}
+			for key, stats := range tsBasicStats {
+				tsMetrics[key] = stats
+			}
+		}
+		tsK8sInfo = append(tsK8sInfo, tsStats)
+	}
+
+	rows := make([]*pb.StatTable_PodGroup_Row, 0)
+
+	for _, split := range tsK8sInfo {
```

Comment:
We essentially do two passes over all the leaves: one in the loop above where we query prometheus and store the results into tsMetrics, and then a second pass here where we build the table rows.  Is this necessary?  Can we combine these loops and build the table rows directly in the first pass?

After calling `getTrafficSplitMetrics` we should have all the information we need to loop over the leaves and construct a table row for each.  I think this allows us to get rid of the `tsK8sInfo` slice entirely.

Reply:
I don't believe we can skip `tsK8sInfo`, because we can return rows for a leaf service that do not have any metrics attached. 

If I create a trafficsplit where `authors` has a weight of `1000m` and `authors-clone` has a weight of `0m` there will be no traffic to `authors-clone` and no Prometheus metrics returned. However we still want to return a row for that leaf service. So we loop through every leaf inside `tsK8sInfo`, construct a row for it, and add the metrics if available.

Reply:
I think we still capture leaves without metrics by looping over backends.  For each backend, we attempt to pull the appropriate metrics out of `tsBasicStats` and then consturct a table row.

Reply:
Cool, am working on this right now.

Reply:
Great suggestion, I refactored that section and we now have just two loops in `trafficSplitResourceQuery`, one through K8s `objects` and one through `backends`.